### PR TITLE
Documentation for OSS rules of play rating

### LINKED ISSUE #533: Add documentation for OssRulesOfPlayRating
- Document the additional command-line options for OSS RoP
- Provide a sample/default configuration file for our rules
- Document/explain our default rules on GitHub pages

### PR DESCRIPTION
**Pull request by SebastianWolf-SAP**
Includes documentation about the new OSS rules of play rating including the relevant command-line options such as `--create-issues`. Fixes #533
Fixes also the file separation issue with the `--data-provider-configs` command-line option.

### CODE REVIEW

File: docs/getting_oss_rules_of_play_rating.md
Comment (sourabhsparkala) [inline docs/getting_oss_rules_of_play_rating.md:166]: [`fosstars-rating-core`](https://github.com/SAP/fosstars-rating-core/tree/fosstars-oss-rules-of-play-sap-report)
- Lets have a common naming convention. In the top most section you use `fosstars-oss-rules-of-play-sap-report`
- Also lets have normal mention instead of an identifier syntax. For example: **Rules of Play report from Fosstars**
- I leave it up to you as how you see fit.
- It would be really nice if the point 2 is applied everywhere.

File: docs/oss_rules_of_play_rating.md
Comment (sourabhsparkala) [inline docs/oss_rules_of_play_rating.md:3]:
```suggestion
This section describes a rules of play rating for OSS projects.
```
Please maintain constant acronyms.

File: docs/oss_rules_of_play_rating.md
Comment (sourabhsparkala) [inline docs/oss_rules_of_play_rating.md:7]: These 2 points should come in the beginning of this section I think.

File: docs/oss_rules_of_play_rating.md
Comment (sourabhsparkala) [inline docs/oss_rules_of_play_rating.md:1]: Also is there some introduction on what `Rule of Play` is?

File: docs/oss_rules_of_play_rating.md
Comment (sourabhsparkala) [inline docs/oss_rules_of_play_rating.md:15]: Before we provide this information on what the rules of play rating should be viewed as, can we provide the list of Rules?

File: docs/oss_rules_of_play_rating.md
Comment (sourabhsparkala) [inline docs/oss_rules_of_play_rating.md:13]:
```suggestion
1.  `PASSED_WITH_WARNING`: the project fulfills the minimum expectations of the rules of play compliance, but shows some issues that should be resolved.
```

File: docs/oss_rules_of_play_rating.md
Comment (sourabhsparkala) [inline docs/oss_rules_of_play_rating.md:14]:
```suggestion
1.  `FAILED`: the project doesn't fulfill the minimum expectation of the rules of play compliance.
```

File: docs/oss_rules_of_play_rating.md
Comment (sourabhsparkala) [reply docs/oss_rules_of_play_rating.md:15]: `UNCLEAR`: Not enough information to reliably calculate a score and a label for the project.

File: docs/oss_rules_of_play_rating.md
Comment (SebastianWolf-SAP) [reply docs/oss_rules_of_play_rating.md]: Done, thanks!

File: docs/oss_rules_of_play_rating.md
Comment (SebastianWolf-SAP) [reply docs/oss_rules_of_play_rating.md:1]: Explanation now comes straight after the header as you proposed it.

File: docs/getting_oss_rules_of_play_rating.md
Comment (sourabhsparkala) [inline docs/getting_oss_rules_of_play_rating.md:7]: - Are there any mandatory iields in the configurations?
- Any special cases to be kept in check?

File: docs/getting_oss_rules_of_play_rating.md
Comment (SebastianWolf-SAP) [reply docs/getting_oss_rules_of_play_rating.md:166]: The first identifier is simply the branch for the SAP report. From my perspective, the more relevant identifier here would be the command-line option `oss-rules-of-play` - that's how Artem called the rating at the first place. That's why I explained everything in the documentation accordingly. 
I'm not against a consistent renaming, but from my perspective also the focus on _open source_ rules of play makes slightly more sense than mentioning that the report comes from Fosstars. Thus, I'd rather keep it as it is right now.

File: docs/getting_oss_rules_of_play_rating.md
Comment (sourabhsparkala) [inline docs/getting_oss_rules_of_play_rating.md:10]:
```suggestion
This data provider `ContributingGuidelineInfo` allows you to specify 2 types of regular expression patterns 
- The content that you expect to be part of contributing guidelines
- The content that you expect must not be part of the contributing guidelines
The option is called `requiredContentPatterns` and expects a list of one or multiple entries. If you enter multiple entries, the contribution guidelines must comply with all the patterns for the rule to pass.
```
May be an example of the pattern?

File: docs/getting_oss_rules_of_play_rating.md
Comment (SebastianWolf-SAP) [reply docs/getting_oss_rules_of_play_rating.md:7]: Yep, for instance the expected license list is mandatory, others can also be omitted. But the details are explained in the respective sub-sections.

File: docs/getting_oss_rules_of_play_rating.md
Comment (sourabhsparkala) [inline docs/getting_oss_rules_of_play_rating.md:13]: The data provider `LicenseInfo`

Comment (sourabhsparkala) [reply docs/getting_oss_rules_of_play_rating.md:13]: There are few redundant lines like 
- `but can also be left empty`.
- Could we split this into steps or points. When we use Finally, it usually means there is few steps above or few options which should be followed.

**Review by sourabhsparkala — COMMENTED** [review]
The comments are just general advice. Overall the document looks really good 👍

File: docs/getting_oss_rules_of_play_rating.md
Comment (SebastianWolf-SAP) [reply docs/getting_oss_rules_of_play_rating.md:16]: Yeah, also thought about bullet points, but for now I kept it like that as it suits a little better to the style of the rest. The "can also be left empty" is there to emphasize that it's an optional entry.

**Comment by SebastianWolf-SAP** [comment]
> The comments are just general advice. Overall the document looks really good 👍

Thank you very much - for the review, feedback and improvement suggestions. 👍

File: docs/getting_oss_rules_of_play_rating.md
Comment (artem-smotrakov) [inline docs/getting_oss_rules_of_play_rating.md:13]:
```suggestion
The data provider `ContributingGuidelineInfo` allows you to specify 2 types of regular expression patterns 
- The content that you expect to be part of the contributing guidelines.
- The content that you expect must not be part of the contributing guidelines.
The option is called `requiredContentPatterns` and expects a list of one or multiple Regex. If you enter multiple entries, the contribution guidelines must comply with all the patterns for the rule to pass.
```

File: docs/getting_oss_rules_of_play_rating.md
Comment (artem-smotrakov) [reply docs/getting_oss_rules_of_play_rating.md:16]:
```suggestion
The data provider `LicenseInfo` allows you to specify which license(s) should be allowed in the analyzed project.
This option is called `allowedLicenses` and expects a list of one or more SPDX license IDs.
In addition you can check for terms that must not in the license (e.g. to avoid modifications) using regular expression patterns.
The corresponding option is called `disallowedLicensePatterns`, and it accepts a list of patterns, but can also be left empty.
Finally, you are also able to specify exceptions, i.e. which repositories shall not be checked with this data provider, for example, because certain repositories are known to use other licenses, but have gone through an exception process.
The option is called `repositoryExceptions` and can accept a list of full repository URLs, e.g. `https://github.com/SAP/fosstars-rating-core`, but can also be left empty if there are no exceptions.
```

File: docs/getting_oss_rules_of_play_rating.md
Comment (artem-smotrakov) [inline docs/getting_oss_rules_of_play_rating.md:19]:
```suggestion
The data provider `ReadmeInfo` allows you to specify which terms or patterns you expect to see in a README file.
The corresponding option `requiredContentPatterns` expects a list of one or more regular expression patterns.
If you enter multiple entries, the README file content must match all patterns for the rule to pass.
```

File: docs/getting_oss_rules_of_play_rating.md
Comment (artem-smotrakov) [inline docs/getting_oss_rules_of_play_rating.md:23]:
```suggestion
In order to be able to identify a certain rule better in the final result report, e.g. for follow-up actions, Fosstars expects that all rules are configured with a customized rule ID. That configuration needs to be in a file called `OssRulesOfPlayRatingMarkdownFormatter.config.yaml` and should be placed in the current working directory. The relevant option is called `ruleIds` and expects a list of key/value pairs where the key is the customized rule ID and the value the descriptive name of the rule that is maintained in the [OssFeatures class](../src/main/java/com/sap/oss/phosphor/fosstars/model/feature/oss/OssFeatures.java). 
```

File: docs/getting_oss_rules_of_play_rating.md
Comment (artem-smotrakov) [inline docs/getting_oss_rules_of_play_rating.md:29]:
```suggestion
The command-line tool can be used to calculate OSS Rules of Play ratings.
```

File: docs/getting_oss_rules_of_play_rating.md
Comment (artem-smotrakov) [inline docs/getting_oss_rules_of_play_rating.md:128]:
```suggestion
The tool prints out the gathered data and a status for the project.
```

File: docs/getting_oss_rules_of_play_rating.md
Comment (artem-smotrakov) [inline docs/getting_oss_rules_of_play_rating.md:131]: This option is not really used by the OSS RoP rating.

```suggestion
```

File: docs/getting_oss_rules_of_play_rating.md
Comment (artem-smotrakov) [inline docs/getting_oss_rules_of_play_rating.md:135]:
```suggestion
Exclusively for the OSS rules of play rating, Fosstars can create GitHub issues for failed rules directly in the affected repositories so the issues can be followed up by the repository owners. If you want to do that, add the command-line option `--create-issues` or enable `ISSUES` reporter in a config file passes through `--config` options.
Fosstars will inform you on the command-line about the issue creation process as well:
```

File: docs/getting_oss_rules_of_play_rating.md
Comment (artem-smotrakov) [inline docs/getting_oss_rules_of_play_rating.md:148]:
```suggestion

Here is an example of a configuration file the scans projects in an organization on GitHub, creates a Markdown report and opens issues for violated rules:

cache: .fosstars/project_rating_cache.json
reports:
  - type: markdown # create a Markdown report in the current directory
    where: ./
  - type: ISSUES # open issues for violated rules
finder:
  organizations: # scan project in the SAP organization
    - name: SAP

```

File: docs/getting_oss_rules_of_play_rating.md
Comment (artem-smotrakov) [inline docs/getting_oss_rules_of_play_rating.md:169]:
```suggestion
SAP already uses the OSS Rules of Play rating in a [GitHub action](https://github.com/SAP/fosstars-rating-core/blob/master/.github/workflows/fosstars-rules-of-play.yml) and posts the results to the branch [fosstars-oss-rules-of-play-sap-report](https://github.com/SAP/fosstars-rating-core/tree/fosstars-oss-rules-of-play-sap-report).
```

Comment (artem-smotrakov) [reply docs/getting_oss_rules_of_play_rating.md:169]:
```suggestion
SAP already uses the OSS Rules of Play rating in a [GitHub action](../.github/workflows/fosstars-rules-of-play.yml) and posts the results in a dedicated branch on [fosstars-rating-core](https://github.com/SAP/fosstars-rating-core/tree/fosstars-oss-rules-of-play-sap-report).
```

**Review by artem-smotrakov — COMMENTED** [review]
Overall, looks good to me @SebastianWolf-SAP !. I left several suggestions, please have a look.

**Comment by SebastianWolf-SAP** [comment]
Thanks for the proposals, @artem-smotrakov - all should be integrated now. I hope, nothing was left out... :)

**Review by artem-smotrakov — APPROVED** [review]
Thanks for addressing the comments @SebastianWolf-SAP ! The changeset looks good to me.

@sourabhsparkala Please merge this pull request if it looks fine to you.

**Comment by sourabhsparkala** [comment]
@SebastianWolf-SAP @artem-smotrakov the changes looks good to me.

**Pull request merged**
